configs/components/pxp-agent.rb

```diff
@@ -82,6 +82,49 @@
     ["#{make} -j$(shell expr $(shell #{platform[:num_cores]}) + 1) install"]
   end
 
+  if platform.is_windows?
+    # In PA-1850, it was found that when there is a system version of openssl
+    # installed on Windows, then the pxp-agent will fail to start properly if
+    # the user tries to restart it. Specifically, Puppet shows the service as
+    # being in the "paused" state. The reason this happened is because when Windows
+    # executes the pxp-agent.exe file, it searches for its required .dll files first
+    # in the .exe file's directory, then in the system library directory
+    # (e.g. C:\Windows:\System32), and then the custom PATH last (as a rough approximation,
+    # it actually searches some other locations, but the relative ordering of ".exe directory"
+    # => "system library directory" => PATH is still there). See https://msdn.microsoft.com/en-us/library/7d83bc18.aspx
+    # for more details. Since we do not have any .dll files in pxp-agent/bin, the system
+    # openssl is detected as pxp-agent's openssl library instead of Puppet's openssl, which
+    # is installed in [INSTALLDIR]\Puppet\puppet\bin. Of course, pxp-agent is not built to work
+    # with the system openssl, so it fails to start properly.
+    #
+    # By copying the dependent .dll files in the puppet/bin directory to pxp-agent/bin,
+    # we ensure that Windows uses our in-house .dll files to start up pxp-agent.
+    #
+    # See https://tickets.puppetlabs.com/browse/PA-1850 for all the details.
+    pkg.install do
+      dependent_dlls = [
```

Comment:
These dependencies were obtained by using http://www.dependencywalker.com/ on the pxp-agent.exe file per @Iristyle's suggestion.

Reply:
What's interesting is that none of the openssl dlls (libeay32.dll and ssleay32.dll) appear in the list of dependencies for the 1.10.x branch.

In the 5.3.x branch, we have these same dependencies plus:
  leatherman_curl.dll
  libstdc++-6.dll
  libeay32.dll

(The other OpenSSL dll, ssleay32.dll, does not appear there).

Reply:
https://github.com/ekinanp/puppet-agent/blob/PA-1850-5.3.x/configs/components/pxp-agent.rb#L93-L106 shows all the dependencies on the 5.3.x branch -- ssleay32.dll is not a dependency there, but I added it anyways since it's an OpenSSL dll.

Reply:
Since this list of dependencies could change in the future, is there a more way to specify this list? I would expect these to be outputs of the OpenSSL build process and referenced as such.

@McdonaldSeanp you know if this is possible in Vanagon?

Reply:
@Iristyle no there's no way in vanagon to output information in one component and consume that info in another component.

We had to make similar changes to this for ruby and we had to do the same thing (where we just list them statically).

TBH I think this is the number 1 reason to try and get https://tickets.puppetlabs.com/browse/PA-761 working for puppet6, because this entire issue would go away.